/***************************************************************************
 *  HandheldFanControl Copyright (C) 2025  Ziperix                         *
 *                                                                         *
 *  This program is free software: you can redistribute it and/or modify   *
 *  it under the terms of the GNU General Public License as published by   *
 *  the Free Software Foundation, either version 3 of the License, or      *
 *  (at your option) any later version.                                    *
 *                                                                         *
 *  This program is distributed in the hope that it will be useful,        *
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of         *
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          *
 *  GNU General Public License for more details.                           *
 *                                                                         *
 *  You should have received a copy of the GNU General Public License      *
 *  along with this program.  If not, see <https://www.gnu.org/licenses/>. *
 ***************************************************************************/

syntax = "proto3";

package hfc.fan_control_service_proto;

message GetPluginVersionRequest {
}

message GetPluginVersionResponse {
  string version = 1;
}

message GetDeviceMetadataRequest {
}

message GetDeviceMetadataResponse {
  string device_name = 1;
  uint64 min_fan_speed = 2;
  uint64 max_fan_speed = 3;
}

message GetCurrentFanSpeedRequest {
}

message GetCurrentFanSpeedResponse {
  uint64 current_fan_speed = 1;
  uint64 user_target_fan_speed = 2;
}

message SetTargetFanSpeedRequest {
  uint64 temperature_celsius = 1;
  uint64 target_fan_speed = 2;
}
